dashboard/app/models/unit.rb

```diff
@@ -64,7 +64,7 @@ class Unit < ApplicationRecord
   belongs_to :wrapup_video, class_name: 'Video', optional: true
   belongs_to :user, optional: true
   has_many :unit_group_units, foreign_key: 'script_id'
-  has_many :unit_groups, through: :unit_group_units
+  has_many :unit_groups, through: :unit_group_units, dependent: :destroy
```

Comment:
please move `dependent: :destroy` to previous line. this looks like it will destroy the unit_group, whereas we want to destroy only the unit_group_unit.

Reply:
Thanks for the catch!! I'll go over the docs to understand why the unit tests were passing.

Reply:
![image](https://github.com/code-dot-org/code-dot-org/assets/124813947/4dc7ff01-ab06-4f58-9041-7b35804b9472)

May be this is why the unit group wasn't getting deleted. I don't fully understand the first part of the sentence, around `belongs_to` through.

Reply:
yes, that could be it! 

> the association on the join model must be a belongs_to

That is referring to these associations: https://github.com/code-dot-org/code-dot-org/blob/757b18217d88bb9f02efb1aabcc028ece0e051d3/dashboard/app/models/unit_group_unit.rb#L22-L23

Reply:
as opposed to has_one, has_many, etc